elastic/kibana
# Configure GraphiteURLs whitelist from kibana.kml

## Summary

This PR disable the configuration of the timelion graphite URL from the advanced settings.
The user can specify a whitelist of URLs in the `kibana.yml` file under `timelion.graphiteUrls` like
```
timelion.graphiteUrls: ['https://graphiteserver1.com', 'https://graphiteserver2.com']
```
By default now we don't have any graphite URL configured. If a whitelist is configured, the default one will be the first on the list.

I've added a check on the `graphite()` function that double checks if the configured URL is on that whitelist. This in the case someone change the URL from the Saved Object API.

To test:
1) configure a white list in the `kibana.dev.yml` file
2) open timelion and use the graphite function `.es(*).graphite()` you will notice an error related to a wrong response from the server
3) update the `timelion:graphite.url` advanced setting using the Saved Object API (from outside kibana dev tool):
```
PUT http://localhost:5601/api/saved_objects/config/8.0.0
{
	"attributes": {
        "timelion:graphite.url":"https://url-not-on-whitelist.com"
	}
}
```
4) reopen the timelion visualization and refresh/update the expression to trigger the graphite function. This time you will see a message that inform that the used URL is not on the whitelist.

**Note**
Every existing user with a configured graphite URL will see that last error message (URL not in the kibana.yml file). This will force them to add the URL into the kibana.yml.


### Checklist

Use ~~strikethroughs~~ to remove checklist items you don't feel are applicable to this PR.


### For maintainers
